xdsl/utils/exceptions.py

```diff
@@ -23,6 +23,10 @@ class VerifyException(DiagnosticException):
     pass
 
 
+class PassFailedException(DiagnosticException):
```

Comment:
Can we please add a docstring? I know it's "captain obvious", but maybe when and where this is supposed to appear/be used? Like in the `InterpretationError` below?